File: scraper.js
#!/usr/bin/env node
/**
 * Web Scraper for julieannbrown.com.au
 * Scrapes images and product information, uploads to Cloudinary
 */

const https = require('https');
const http = require('http');
const fs = require('fs');
const path = require('path');
const { URL } = require('url');

// Configuration
const CLOUDINARY_CLOUD = 'fqtug4al';
const CLOUDINARY_PRESET = 'julieannbrown-website';
const OLD_WEBSITE = 'https://julieannbrown.com.au';
const DOWNLOAD_DIR = path.join(__dirname, '.scraper-downloads');

// Ensure download directory exists
if (!fs.existsSync(DOWNLOAD_DIR)) {
  fs.mkdirSync(DOWNLOAD_DIR, { recursive: true });
}

// ─────────────────────────────────────────────────────────────────
// Utility Functions
// ─────────────────────────────────────────────────────────────────

function fetchUrl(url) {
  return new Promise((resolve, reject) => {
    const protocol = url.startsWith('https') ? https : http;
    const timeoutMs = 10000;
    
    const req = protocol.get(url, { 
      timeout: timeoutMs,
      headers: { 'User-Agent': 'Mozilla/5.0' }
    }, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve(data));
    });
    
    req.on('error', reject);
    req.on('timeout', () => {
      req.destroy();
      reject(new Error(`Request timeout for ${url}`));
    });
  });
}

function downloadFile(url, filename) {
  return new Promise((resolve, reject) => {
    const protocol = url.startsWith('https') ? https : http;
    const filepath = path.join(DOWNLOAD_DIR, filename);
    const file = fs.createWriteStream(filepath);
    
    const req = protocol.get(url, { 
      headers: { 'User-Agent': 'Mozilla/5.0' }
    }, (res) => {
      res.pipe(file);
      file.on('finish', () => {
        file.close();
        resolve(filepath);
      });
    });
    
    req.on('error', () => {
      fs.unlink(filepath, () => {});
      reject(new Error(`Failed to download ${url}`));
    });
    file.on('error', reject);
  });
}

async function uploadToCloudinary(filePath, folder) {
  const FormData = require('form-data');
  const fs = require('fs');
  
  const form = new FormData();
  form.append('file', fs.createReadStream(filePath));
  form.append('upload_preset', CLOUDINARY_PRESET);
  form.append('folder', folder);
  
  return new Promise((resolve, reject) => {
    const url = `https://api.cloudinary.com/v1_1/${CLOUDINARY_CLOUD}/image/upload`;
    const request = https.request(url, {
      method: 'POST',
      headers: form.getHeaders(),
      timeout: 30000
    }, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        try {
          const json = JSON.parse(data);
          resolve(json.secure_url);
        } catch (e) {
          reject(new Error(`Invalid response from Cloudinary: ${data}`));
        }
      });
    });
    
    request.on('error', reject);
    request.on('timeout', () => {
      request.destroy();
      reject(new Error('Upload timeout'));
    });
    
    form.pipe(request);
  });
}

// ─────────────────────────────────────────────────────────────────
// HTML Parsing
// ─────────────────────────────────────────────────────────────────

function extractImageSrc(html, baseUrl) {
  const regex = /<img[^>]*src=['"]([^'"]+)['"]/gi;
  const images = [];
  let match;
  
  while ((match = regex.exec(html)) !== null) {
    let src = match[1];
    
    // Skip data URIs and invalid URLs
    if (src.startsWith('data:') || src.startsWith('#')) continue;
    
    // Convert relative URLs to absolute
    if (!src.startsWith('http')) {
      src = new URL(src, baseUrl).href;
    }
    
    images.push(src);
  }
  
  return [...new Set(images)]; // Remove duplicates
}

function extractImageAlt(html) {
  const alts = {};
  const regex = /<img[^>]*src=['"]([^'"]+)['"][^>]*alt=['"]([^'"]+)['"]/gi;
  let match;
  
  while ((match = regex.exec(html)) !== null) {
    alts[match[1]] = match[2];
  }
  
  return alts;
}

function extractLinks(html, baseUrl) {
  const regex = /<a[^>]*href=['"]([^'"]+)['"]/gi;
  const links = [];
  let match;
  
  while ((match = regex.exec(html)) !== null) {
    let href = match[1];
    if (href.startsWith('http')) {
      links.push(href);
    } else if (!href.startsWith('#') && !href.startsWith('javascript:')) {
      links.push(new URL(href, baseUrl).href);
    }
  }
  
  return links;
}

// ─────────────────────────────────────────────────────────────────
// Scraper Functions
// ─────────────────────────────────────────────────────────────────

async function scrapePage(url) {
  try {
    console.log(`📄 Scraping: ${url}`);
    const html = await fetchUrl(url);
    return { url, html };
  } catch (error) {
    console.error(`❌ Error scraping ${url}: ${error.message}`);
    return null;
  }
}

async function scrapeGallery() {
  console.log('\n🎨 SCRAPING GALLERY...');
  const pages = [
    `${OLD_WEBSITE}/portfolio`,
    `${OLD_WEBSITE}/gallery`,
    `${OLD_WEBSITE}/`,
  ];
  
  const gallery = [];
  const processed = new Set();
  
  for (const pageUrl of pages) {
    const page = await scrapePage(pageUrl);
    if (!page) continue;
    
    const images = extractImageSrc(page.html, pageUrl);
    const alts = extractImageAlt(page.html);
    
    for (const imgSrc of images) {
      if (processed.has(imgSrc)) continue;
      processed.add(imgSrc);
      
      // Only include WordPress uploads
      if (!imgSrc.includes('/wp-content/uploads/')) continue;
      
      // Filter out tracking pixels and icons
      if (imgSrc.includes('pixel') || imgSrc.includes('favicon') || 
          imgSrc.includes('icon') || imgSrc.includes('logo') ||
          imgSrc.includes('shopping-bag') || imgSrc.includes('.svg')) continue;
      
      const title = alts[imgSrc] || path.basename(imgSrc).split('.')[0];
      
      gallery.push({
        url: imgSrc,
        title: title,
        category: 'Landscape' // Default, could be improved
      });
    }
  }
  
  return gallery;
}

async function scrapeShop() {
  console.log('\n🛍️  SCRAPING SHOP...');
  const pageUrl = `${OLD_WEBSITE}/shop`;
  
  const products = [];
  const processed = new Set();
  
  const page = await scrapePage(pageUrl);
  if (!page) return products;
  
  // Extract WooCommerce product data from HTML
  const productRegex = /<li[^>]*class="[^"]*product[^"]*"[^>]*>(.*?)<\/li>/gs;
  const matches = page.html.matchAll(productRegex);
  
  for (const match of matches) {
    const productHtml = match[1];
    
    // Extract image
    const imgMatch = productHtml.match(/<img[^>]*src="([^"]+)"/);
    if (!imgMatch) continue;
    const imgSrc = imgMatch[1];
    if (processed.has(imgSrc)) continue;
    processed.add(imgSrc);
    
    // Skip if not a WordPress upload
    if (!imgSrc.includes('/wp-content/uploads/')) continue;
    
    // Extract product title
    const titleMatch = productHtml.match(/<h2[^>]*class="[^"]*product[^"]*"[^>]*>([^<]+)<\/h2>/);
    const title = titleMatch ? titleMatch[1].trim() : 'Product';
    
    // Extract price
    const priceMatch = productHtml.match(/<span[^>]*class="[^"]*price[^"]*"[^>]*>.*?([0-9]+(?:\.[0-9]{2})?)/);
    const price = priceMatch ? parseFloat(priceMatch[1]) : null;
    
    // Extract product link
    const linkMatch = productHtml.match(/<a[^>]*href="([^"]+)"[^>]*class="[^"]*woocommerce[^"]*"/);
    const productLink = linkMatch ? linkMatch[1] : null;
    
    products.push({
      url: imgSrc,
      title: title,
      price: price,
      medium: null,
      productPageUrl: productLink
    });
  }
  
  // Scrape product pages for details
  console.log(`\n📖 Fetching details for ${products.length} products...`);
  for (let i = 0; i < products.length; i++) {
    if (!products[i].productPageUrl) continue;
    
    try {
      console.log(`  [${i+1}/${products.length}] Fetching ${products[i].title}...`);
      const productPage = await scrapePage(products[i].productPageUrl);
      if (!productPage) continue;
      
      // Try to extract medium from product description
      const mediumMatch = productPage.html.match(/<p[^>]*>.*?(?:Medium|Material):\s*([^<]+)<\/p>/i);
      if (mediumMatch) {
        products[i].medium = mediumMatch[1].trim();
      }
      
      // Look for medium in description text
      const descMatch = productPage.html.match(/<div[^>]*class="[^"]*description[^"]*"[^>]*>(.*?)<\/div>/s);
      if (descMatch && !products[i].medium) {
        const text = descMatch[1].replace(/<[^>]+>/g, ' ').substring(0, 200);
        if (text.includes('oil') || text.includes('acrylic') || text.includes('watercolor') ||
            text.includes('canvas') || text.includes('paper') || text.includes('mixed')) {
          products[i].medium = text;
        }
      }
      
      await new Promise(r => setTimeout(r, 500)); // Rate limit
    } catch (error) {
      console.error(`    ⚠️  Could not fetch details: ${error.message}`);
    }
  }
  
  return products;
}

// ─────────────────────────────────────────────────────────────────
// Main Scraping Flow
// ─────────────────────────────────────────────────────────────────

async function main() {
  console.log('🚀 Starting julieannbrown.com.au scraper...\n');
  
  try {
    // Check if form-data is installed
    try {
      require('form-data');
    } catch (e) {
      console.log('📦 Installing required dependencies...');
      const { execSync } = require('child_process');
      execSync('npm install form-data', { stdio: 'inherit', cwd: __dirname });
    }
    
    const gallery = await scrapeGallery();
    const shop = await scrapeShop();
    
    console.log(`\n✅ Found ${gallery.length} gallery images`);
    console.log(`✅ Found ${shop.length} shop products`);
    
    if (gallery.length === 0 && shop.length === 0) {
      console.error('\n⚠️  No content found. The website may have a different structure.');
      console.log('💡 Next steps:');
      console.log('1. Check if julieannbrown.com.au is accessible');
      console.log('2. Inspect the HTML structure manually');
      console.log('3. Update the scraper to match the actual website structure');
      process.exit(1);
    }
    
    console.log('\n📥 Downloading images...');
    
    const galleryData = [];
    for (let i = 0; i < gallery.length; i++) {
      const item = gallery[i];
      try {
        const ext = path.extname(new URL(item.url).pathname) || '.jpg';
        const filename = `gallery-${i}${ext}`;
        console.log(`  [${i+1}/${gallery.length}] ${item.title}`);
        
        const localPath = await downloadFile(item.url, filename);
        console.log(`    ✓ Downloaded to ${filename}`);
        
        console.log(`    ⏳ Uploading to Cloudinary...`);
        const cloudinaryUrl = await uploadToCloudinary(localPath, 'gallery');
        console.log(`    ✓ Uploaded: ${cloudinaryUrl}`);
        
        galleryData.push({
          id: 'g' + Date.now() + i,
          title: item.title,
          src: cloudinaryUrl,
          category: item.category || 'Gallery',
          featured: i === 0 // Make first one featured
        });
        
        // Clean up local file
        fs.unlinkSync(localPath);
        
        // Rate limit to avoid overwhelming APIs
        await new Promise(r => setTimeout(r, 1000));
      } catch (error) {
        console.error(`    ❌ Failed: ${error.message}`);
      }
    }
    
    const shopData = [];
    for (let i = 0; i < shop.length; i++) {
      const item = shop[i];
      try {
        const ext = path.extname(new URL(item.url).pathname) || '.jpg';
        const filename = `shop-${i}${ext}`;
        console.log(`  [${i+1}/${shop.length}] ${item.title}`);
        
        const localPath = await downloadFile(item.url, filename);
        console.log(`    ✓ Downloaded to ${filename}`);
        
        console.log(`    ⏳ Uploading to Cloudinary...`);
        const cloudinaryUrl = await uploadToCloudinary(localPath, 'shop');
        console.log(`    ✓ Uploaded: ${cloudinaryUrl}`);
        
        shopData.push({
          id: 's' + Date.now() + i,
          title: item.title,
          src: cloudinaryUrl,
          price: item.price || 0,
          medium: item.medium || '',
          sold: false,
          stripeLink: '',
          order: i
        });
        
        fs.unlinkSync(localPath);
        await new Promise(r => setTimeout(r, 1000));
      } catch (error) {
        console.error(`    ❌ Failed: ${error.message}`);
      }
    }
    
    // Save data files
    console.log('\n💾 Saving data files...');
    fs.writeFileSync(
      path.join(__dirname, 'data', 'gallery.json'),
      JSON.stringify(galleryData, null, 2)
    );
    console.log('✓ Saved data/gallery.json');
    
    fs.writeFileSync(
      path.join(__dirname, 'data', 'shop.json'),
      JSON.stringify(shopData, null, 2)
    );
    console.log('✓ Saved data/shop.json');
    
    // Clean up
    console.log('\n🧹 Cleaning up...');
    fs.rmSync(DOWNLOAD_DIR, { recursive: true, force: true });
    
    console.log('\n✨ Done! Your website has been updated with:');
    console.log(`   • ${galleryData.length} gallery images`);
    console.log(`   • ${shopData.length} shop products`);
    console.log('\n📋 Next steps:');
    console.log('   1. Review data/gallery.json and data/shop.json');
    console.log('   2. Add missing product details (prices, mediums, stripe links)');
    console.log('   3. Commit changes to GitHub');
    
  } catch (error) {
    console.error('❌ Fatal error:', error.message);
    process.exit(1);
  }
}

main().catch(error => {
  console.error('❌ Unhandled error:', error);
  process.exit(1);
});

File: scraper.py
#!/usr/bin/env python3
"""
Web scraper for julieannbrown.com.au
Scrapes images and product information, uploads to Cloudinary
"""

import urllib.request
import urllib.error
import json
import re
import os
import sys
from pathlib import Path
from urllib.parse import urljoin, urlparse
import time

# Configuration
CLOUDINARY_CLOUD = 'fqtug4al'
CLOUDINARY_PRESET = 'julieannbrown-website'
OLD_WEBSITE = 'https://julieannbrown.com.au'
SCRIPT_DIR = Path(__file__).parent
DOWNLOAD_DIR = SCRIPT_DIR / '.scraper-downloads'
DATA_DIR = SCRIPT_DIR / 'data'

# Ensure directories exist
DOWNLOAD_DIR.mkdir(exist_ok=True)
DATA_DIR.mkdir(exist_ok=True)

# ─────────────────────────────────────────────────────────────────
# Utility Functions
# ─────────────────────────────────────────────────────────────────

def fetch_url(url, timeout=10):
    """Fetch URL content"""
    try:
        req = urllib.request.Request(
            url,
            headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'}
        )
        with urllib.request.urlopen(req, timeout=timeout) as response:
            return response.read().decode('utf-8', errors='ignore')
    except Exception as e:
        print(f"  ❌ Error fetching {url}: {str(e)}")
        return None

def extract_images(html, base_url):
    """Extract image URLs from HTML"""
    images = []
    # Find all img src attributes
    pattern = r'src=[\'"]([^\'"]+)[\'"]'
    matches = re.findall(pattern, html)
    
    seen = set()
    for src in matches:
        # Skip data URIs and invalid URLs
        if src.startswith('data:') or src.startswith('#'):
            continue
        
        # Convert relative URLs to absolute
        if not src.startswith('http'):
            src = urljoin(base_url, src)
        
        if src not in seen:
            images.append(src)
            seen.add(src)
    
    return images

def extract_product_info(html):
    """Extract product information from shop HTML"""
    products = []
    
    # Find product list items
    pattern = r'<li[^>]*class="[^"]*product[^"]*"[^>]*>(.*?)<\/li>'
    matches = re.finditer(pattern, html, re.DOTALL)
    
    for match in matches:
        product_html = match.group(1)
        
        # Extract image
        img_match = re.search(r'<img[^>]*src="([^"]+)"', product_html)
        if not img_match:
            continue
        img_src = img_match.group(1)
        
        # Skip if not a WordPress upload
        if '/wp-content/uploads/' not in img_src:
            continue
        
        # Extract title
        title_match = re.search(r'<h2[^>]*class="[^"]*product[^"]*"[^>]*>([^<]+)<\/h2>', product_html)
        title = title_match.group(1).strip() if title_match else 'Product'
        
        # Extract price
        price_match = re.search(r'<span[^>]*class="[^"]*price[^"]*"[^>]*>.*?([0-9]+(?:\.[0-9]{2})?)', product_html)
        price = float(price_match.group(1)) if price_match else None
        
        # Extract product link
        link_match = re.search(r'<a[^>]*href="([^"]+)"[^>]*class="[^"]*woocommerce[^"]*"', product_html)
        product_url = link_match.group(1) if link_match else None
        
        products.append({
            'url': img_src,
            'title': title,
            'price': price,
            'medium': None,
            'productPageUrl': product_url
        })
    
    return products

def download_file(url, filename):
    """Download file from URL"""
    try:
        filepath = DOWNLOAD_DIR / filename
        req = urllib.request.Request(
            url,
            headers={'User-Agent': 'Mozilla/5.0'}
        )
        with urllib.request.urlopen(req, timeout=10) as response:
            with open(filepath, 'wb') as f:
                f.write(response.read())
        return filepath
    except Exception as e:
        print(f"    ❌ Download failed: {str(e)}")
        return None

def upload_to_cloudinary(file_path, folder):
    """Upload file to Cloudinary using curl"""
    import subprocess
    
    try:
        cmd = [
            'curl',
            '-X', 'POST',
            '-F', f'file=@{file_path}',
            '-F', f'upload_preset={CLOUDINARY_PRESET}',
            '-F', f'folder={folder}',
            f'https://api.cloudinary.com/v1_1/{CLOUDINARY_CLOUD}/image/upload'
        ]
        result = subprocess.run(cmd, capture_output=True, text=True, timeout=30)
        
        if result.returncode == 0:
            response = json.loads(result.stdout)
            return response.get('secure_url')
        else:
            print(f"    ❌ Upload failed: {result.stderr}")
            return None
    except Exception as e:
        print(f"    ❌ Upload error: {str(e)}")
        return None

# ─────────────────────────────────────────────────────────────────
# Scraping Functions
# ─────────────────────────────────────────────────────────────────

def scrape_gallery():
    """Scrape gallery images from website"""
    print('\n🎨 SCRAPING GALLERY...')
    pages = [
        f'{OLD_WEBSITE}/',
        f'{OLD_WEBSITE}/portfolio',
        f'{OLD_WEBSITE}/gallery',
    ]
    
    gallery = []
    processed = set()
    
    for page_url in pages:
        print(f"  📄 Fetching {page_url}...")
        html = fetch_url(page_url)
        if not html:
            continue
        
        images = extract_images(html, page_url)
        
        for img_src in images:
            if img_src in processed:
                continue
            processed.add(img_src)
            
            # Only include WordPress uploads
            if '/wp-content/uploads/' not in img_src:
                continue
            
            # Filter out unneeded images
            if any(x in img_src.lower() for x in ['pixel', 'favicon', 'icon', 'logo', 'shopping-bag', '.svg']):
                continue
            
            title = os.path.basename(img_src).split('.')[0]
            gallery.append({
                'url': img_src,
                'title': title,
                'category': 'Landscape'
            })
    
    print(f"  ✓ Found {len(gallery)} gallery images")
    return gallery

def scrape_shop():
    """Scrape shop products from website"""
    print('\n🛍️  SCRAPING SHOP...')
    page_url = f'{OLD_WEBSITE}/shop'
    
    print(f"  📄 Fetching {page_url}...")
    html = fetch_url(page_url)
    if not html:
        return []
    
    products = extract_product_info(html)
    print(f"  ✓ Found {len(products)} shop products")
    
    # Fetch product pages for details
    if products:
        print(f'\n  📖 Fetching details for products...')
        for i, product in enumerate(products):
            if not product['productPageUrl']:
                continue
            
            print(f"    [{i+1}/{len(products)}] {product['title']}...")
            product_html = fetch_url(product['productPageUrl'])
            
            if product_html:
                # Try to extract medium from product description
                medium_match = re.search(r'<p[^>]*>.*?(?:Medium|Material):\s*([^<]+)<\/p>', product_html, re.IGNORECASE)
                if medium_match:
                    product['medium'] = medium_match.group(1).strip()
                
                # Look for medium keywords in description
                if not product['medium']:
                    desc_match = re.search(r'<div[^>]*class="[^"]*description[^"]*"[^>]*>(.*?)<\/div>', product_html, re.DOTALL)
                    if desc_match:
                        text = re.sub(r'<[^>]+>', ' ', desc_match.group(1)).strip()[:200]
                        if any(x in text.lower() for x in ['oil', 'acrylic', 'watercolor', 'canvas', 'paper', 'mixed']):
                            product['medium'] = text
            
            time.sleep(0.5)  # Rate limit
    
    return products

# ─────────────────────────────────────────────────────────────────
# Main
# ─────────────────────────────────────────────────────────────────

def main():
    print('🚀 Starting julieannbrown.com.au scraper...')
    
    try:
        # Check if curl is available
        import subprocess
        subprocess.run(['curl', '--version'], capture_output=True, check=True)
    except:
        print('❌ curl is required but not found. Please install curl.')
        sys.exit(1)
    
    gallery = scrape_gallery()
    shop = scrape_shop()
    
    if gallery is None:
        gallery = []
    if shop is None:
        shop = []
    
    print(f'\n✅ Found {len(gallery)} gallery images')
    print(f'✅ Found {len(shop)} shop products')
    
    if len(gallery) == 0 and len(shop) == 0:
        print('\n⚠️  No content found. Please check the website structure.')
        sys.exit(1)
    
    # Download and upload images
    print('\n📥 Downloading and uploading images...')
    
    gallery_data = []
    for i, item in enumerate(gallery):
        try:
            print(f'  [{i+1}/{len(gallery)}] {item["title"][:50]}...')
            
            # Download
            ext = os.path.splitext(urlparse(item['url']).path)[1] or '.jpg'
            filename = f'gallery-{i}{ext}'
            file_path = download_file(item['url'], filename)
            
            if file_path:
                print(f'    ✓ Downloaded')
                
                # Upload
                print(f'    ⏳ Uploading to Cloudinary...')
                cloudinary_url = upload_to_cloudinary(file_path, 'gallery')
                
                if cloudinary_url:
                    print(f'    ✓ Uploaded')
                    gallery_data.append({
                        'id': f'g{int(time.time() * 1000) + i}',
                        'title': item['title'],
                        'src': cloudinary_url,
                        'category': item['category'],
                        'featured': i == 0
                    })
                    os.remove(file_path)
                    time.sleep(1)
        except Exception as e:
            print(f'    ❌ Error: {str(e)}')
    
    shop_data = []
    for i, item in enumerate(shop):
        try:
            print(f'  [{i+1}/{len(shop)}] {item["title"][:50]}...')
            
            # Download
            ext = os.path.splitext(urlparse(item['url']).path)[1] or '.jpg'
            filename = f'shop-{i}{ext}'
            file_path = download_file(item['url'], filename)
            
            if file_path:
                print(f'    ✓ Downloaded')
                
                # Upload
                print(f'    ⏳ Uploading to Cloudinary...')
                cloudinary_url = upload_to_cloudinary(file_path, 'shop')
                
                if cloudinary_url:
                    print(f'    ✓ Uploaded')
                    shop_data.append({
                        'id': f's{int(time.time() * 1000) + i}',
                        'title': item['title'],
                        'src': cloudinary_url,
                        'price': item['price'] or 0,
                        'medium': item['medium'] or '',
                        'sold': False,
                        'stripeLink': '',
                        'order': i
                    })
                    os.remove(file_path)
                    time.sleep(1)
        except Exception as e:
            print(f'    ❌ Error: {str(e)}')
    
    # Save data files
    print('\n💾 Saving data files...')
    with open(DATA_DIR / 'gallery.json', 'w') as f:
        json.dump(gallery_data, f, indent=2)
    print('✓ Saved data/gallery.json')
    
    with open(DATA_DIR / 'shop.json', 'w') as f:
        json.dump(shop_data, f, indent=2)
    print('✓ Saved data/shop.json')
    
    # Cleanup
    print('\n🧹 Cleaning up...')
    import shutil
    if DOWNLOAD_DIR.exists():
        shutil.rmtree(DOWNLOAD_DIR)
    
    print('\n✨ Done! Your website has been updated with:')
    print(f'   • {len(gallery_data)} gallery images')
    print(f'   • {len(shop_data)} shop products')
    print('\n📋 Next steps:')
    print('   1. Review data/gallery.json and data/shop.json')
    print('   2. Add missing product details (prices, mediums, stripe links)')
    print('   3. Commit changes to GitHub')

if __name__ == '__main__':
    main()
